overlord/snapstate/snapmgr.go

```diff
@@ -484,6 +492,18 @@ func (m *SnapManager) ensureRefreshes() error {
 		return err
 	}
 
+	delta := timeutil.Next(refreshSchedule, lastRefresh)
+
+	m.nextRefresh = time.AfterFunc(delta, func() { m.doAutoRefresh(updated, tasksets) })
```

Comment:
If we have a nextRefresh setup and the refreshSchedule string has not changed the function exists before. But maybe it should be done differently so that the control flow is easier to follow?